app/javascript/components/tree-view/hierarchical-tree-view.jsx

```diff
@@ -1,72 +1,168 @@
+/* eslint camelcase: ["warn", {allow: ["bs_tree", "tree_name", "click_url", "check_url", "allow_reselect", "hierarchical_check", "silent_activate", "select_node"]}] */
 import React, { useEffect } from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import Tree, {
   Node,
-  defaultStore,
-  callBack,
-  ActionTypes,
 } from '@manageiq/react-ui-components/dist/wooden-tree';
+import { http } from '../../http_api';
+import { combineReducers } from '../../helpers/redux';
+import reducers, { ACTIONS } from './reducers/index';
+import basicStore from './reducers/basicStore';
 
-const convertData = data => Tree.convertHierarchicalTree(Tree.initHierarchicalTree(JSON.parse(data)));
+const flatten = data => Tree.convertHierarchicalTree(Tree.initHierarchicalTree(data));
+
+const callBack = (nodeId, type, value, namespace) => ({
+  nodeId, type, value, namespace,
+});
+
+/**
+ * Helper
+ * Changes class to className.
+ * Sets undefined checkbox to null.
+ * Moves the key to attr.key.
+ */
+const convert = tree => tree.map((n) => {
+  let node = { ...n };
+  if (node.class) {
+    node = { ...node, classes: node.class };
+    delete node.class;
+  }
+
+  if (node.state.checked === 'undefined') {
```

Comment:
`undefined` as a string? Is that intentional? Or a remnant of a previous typeof?

Reply:
This is intentional, unfortunately ruby doesn't know `undefined` as a type :confused:

Reply:
^

Reply:
Maybe all we really need is a comment :)